cmd/utils.go

```diff
@@ -3,44 +3,58 @@ package cmd
 import (
 	"fmt"
 	"os"
-	"path/filepath"
-	"strings"
+
+	"github.com/bitrise-core/bitrise-init/scanners/ios"
+	"github.com/bitrise-core/bitrise-init/scanners/xamarin"
+	"github.com/bitrise-core/bitrise-init/utility"
 )
 
-// ProjectType ...
-type ProjectType string
+// IDEType ...
+type IDEType string
 
 const (
-	// iOS
-	xcodeWorkspace ProjectType = "*.xcworkspace"
-	xcodeProject   ProjectType = "*.xcodeproj"
+	xcodeIDE   IDEType = "iOS"
+	xamarinIDE IDEType = "xamarin"
 )
 
 // Scans the root dir for the provided project files
-// If none of them in the root dir, then it will return an error
-func scanForProjectFiles(projectFiles []ProjectType) (string, error) {
-	root, err := os.Getwd()
+func scanForProjectFiles(ideType IDEType) ([]string, error) {
+	searchDir, err := os.Getwd()
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
-	for _, projectFile := range projectFiles {
-		pathPattern := filepath.Join(root, string(projectFile))
-		paths, err := filepath.Glob(pathPattern)
-		if err != nil {
-			return "", err
-		}
+	fileList, err := utility.ListPathInDirSortedByComponents(searchDir, false)
+	if err != nil {
+		return nil, fmt.Errorf("failed to search for files in (%s), error: %s", searchDir, err)
+	}
 
-		switch len(paths) {
-		case 0:
-			continue
-		case 1:
-			return paths[0], nil
-		default:
-			return "", fmt.Errorf("multiple project file (%s) found in the root (%s) directory: %s", projectFile, root, strings.Join(paths, "\n"))
+	var paths []string
+	{
+		if ideType == xcodeIDE {
+			paths, err = ios.FilterRelevantWorkspaceFiles(fileList)
+			if err != nil {
+				return nil, fmt.Errorf("failed to search for solution files, error: %s", err)
```

Comment:
solution -> workspace

Reply:
Fixed in:
https://github.com/bitrise-tools/codesigndoc/pull/95/commits/5311596c354bbeaf882314e63324e48b9a0df343